server/mscalendar/oauth2.go

```diff
@@ -123,9 +123,15 @@ func (app *oauth2App) CompleteOAuth2(authedUserID, code, state string) error {
 
 	if mattermostUserID == app.Config.BotUserID {
 		app.Poster.DM(authedUserID, BotWelcomeMessage, me.Mail)
-	} else {
-		app.Poster.DM(mattermostUserID, WelcomeMessage, me.Mail)
+		return nil
 	}
 
+	err = app.Store.StoreUserInIndex(u)
```

Comment:
Another potential problem here is... What if I cannot store the index? What happens with the "detached user" (a user stored in the KVStore but not referenced in the index)?

My first thought is that will not be a big problem, since the user will try to connect again, and will overwrite the value, but I wonder what are your thoughts on that.

Reply:
Hmmm I'm not sure. I believe any operations trying to access the user from the index will gracefully fail (meaning no panic). I think we should tell the user to try again from here. I'm actually not sure what the behavior is to the user when they hit an error here.

Reply:
My concern is more about leaving "thrash" on the KVStore, but probably not a real problem.

Reply:
Right, almost like we need a `transaction` for multiple KV keys. Using a `cluster.Mutex` and rollback functionality, we can implement it. @levb and I have talked about this. We would need to design it into the `store` or `utils/kvstore` package.